cmd/restic/cmd_prune.go

```diff
@@ -297,19 +299,51 @@ func pruneRepository(gopts GlobalOptions, repo restic.Repository) error {
 	}
 
 	if len(removePacks) != 0 {
-		bar = newProgressMax(!gopts.Quiet, uint64(len(removePacks)), "packs deleted")
-		bar.Start()
-		for packID := range removePacks {
-			h := restic.Handle{Type: restic.DataFile, Name: packID.String()}
-			err = repo.Backend().Remove(ctx, h)
-			if err != nil {
-				Warnf("unable to remove file %v from the repository\n", packID.Str())
-			}
-			bar.Report(restic.Stat{Blobs: 1})
-		}
-		bar.Done()
+		DeleteFiles(gopts, true, repo, removePacks, restic.DataFile)
 	}
 
 	Verbosef("done\n")
 	return nil
 }
+
+const numDeleteWorkers = 8
+
+// DeleteFiles deletes the given fileList of fileType in parallel
+// if ignoreError=true, it will print a warning if there was an error, else it will abort.
+func DeleteFiles(gopts GlobalOptions, ignoreError bool, repo restic.Repository, fileList restic.IDSet, fileType restic.FileType) error {
```

Comment:
I think it would be a good idea to replace the `ignoreError` flag with `DeleteFiles` and `DeleteFilesChecked` functions, which then internally can call a third function which accepts that flag.

Reply:
That is a good idea! Is changed.